src/test/java/pages/BasePage.java

```diff
@@ -15,26 +15,68 @@ public BasePage(WebDriver driver) {
         PageFactory.initElements(driver, this);
     }
 
+
     protected void type(WebElement fieldWebElement, String text) {
         waitForElementToBeClickable(fieldWebElement);
         fieldWebElement.click();
         fieldWebElement.sendKeys(text);
     }
 
+    protected void clearAndType(WebElement fieldWebElement, String text) {
+        waitForElementToBeClickable(fieldWebElement);
+        fieldWebElement.click();
+        fieldWebElement.clear();
+        fieldWebElement.sendKeys(text);
+    }
+
     protected void scrollToElementAndClick(WebElement webElement) {
         JavascriptExecutor js = (JavascriptExecutor) driver;
         js.executeScript("window.scrollBy(0,4000)");
         waitForElementToBeClickable(webElement);
         webElement.click();
     }
 
+    protected void scrollToElement(WebElement webElement, int px) {
+        // todo: rewrite to optimise the scroll for all screen
```

Comment:
Usually all the comments have to be removed but only because it's your study proj and not portfolio, lgtm.